### PR TITLE
feat(solidity): use maintained parser and update highlights accordingly

### LINKED ISSUE #3421: Switch to upstream solidity parser
Looks like the current repo used for solidity (https://github.com/YongJieYongJie/tree-sitter-solidity) isn't being maintained. I think it makes sense to switch to the upstream repo (https://github.com/JoranHonig/tree-sitter-solidity). 

I've made the changes in the following commit and I can create a PR
https://github.com/shuklaayush/nvim-treesitter/commit/fc4c6db7119f86971f9024cb5c3a5c46bdf64ffe

cc @YongJieYongJie

### PR DESCRIPTION
**Pull request by amaanq**
Closes #3421 (cc: @shuklaayush)

This PR uses the maintained parser (of which the current nvim-treesitter parser is a fork of, and now somewhat dead), and updates the highlights accordingly

### CODE REVIEW

**Review by theHamsta — APPROVED** [review]

**Comment by amaanq** [comment]
will fix the failing ci later today

File: queries/solidity/highlights.scm [hunk 1]
```diff
@@ -81,100 +80,122 @@
...
+
+[ "[" "]" ] @punctuation.bracket
 
+[ "(" ")" ] @punctuation.bracket
 
 [
   "."
   ","
+	; FIXME: update grammar
+	; (semicolon)
```
Comment (ObserverOfTime) [inline queries/solidity/highlights.scm:186]:
```suggestion
  ; FIXME: update grammar
  ; (semicolon)
```

File: queries/solidity/highlights.scm [hunk 2]
```diff
@@ -81,100 +80,122 @@
...
- "view"
- "payable"
+  "public"
+  "internal"
+  "private"
+  "external"
+  "pure"
+  "view"
+  "payable"
+	(immutable)
```
Comment (ObserverOfTime) [inline queries/solidity/highlights.scm:122]:
```suggestion
  (immutable)
```

**Review by ObserverOfTime — APPROVED** [review]

File: queries/solidity/highlights.scm (hunk 2 above)
Comment (amaanq) [reply queries/solidity/highlights.scm]: oops, thanks

**Comment by amaanq** [comment]
ci fixed

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**